### ISSUE TITLE
Email field for authors accepts text rather than email

### ISSUE THREAD

**Issue by kmccurley** [kohler/hotcrp#206]
In most web forms, a form that asks for an email would have `<input type="email">` in order to use client-side validation. Unfortunately hotcrp uses `type="text"`, and we found that some authors entered information like an affiliation into that field. It should be an easy change to impose client-side validation on email addresses so that this doesn't happen in the future.

**Comment by kohler** [comment]
This is on purpose for several reasons including that validation is generally nasty and annoying. Which email input are you thinking of here? There's some code in the backend that attempts to validate email fields for “affiliation-ness” and DTRT.

**Comment by kmccurley** [comment]
Specifically the field auemail1 in the submission form captures information for the authorInformation field of the Paper table, and this field is not validated to contain an email address.

**Comment by kohler** [comment]
I don't know what version you're using and there have been changes here. In mainline the field is called authors:email_1 not auemail1 and it is validated server side.

**Comment by kmccurley** [comment]
We have been reluctant to pull in updates because there doesn't appear to be a release cycle. The last release appears to be from August 2018. If this is fixed in the current head version then you should just close the issue.

**Comment by kohler** [comment]
Yes, this is an issue. HotCRP has gone through a lot of internal changes in the last two years. I expect to make another release soon. Though it will have a version number jump, it should be safe to apply in the usual way. Have you modified the code at all?

**Comment by kmccurley** [comment]
We have been trying to confine our changes to a separate directory and modify as little as possible in the core HotCRP code, but changes to authentication or what is stored in database fields could break our extensions. We decided to keep our extensions external in part because of the way HotCRP pages are generated, which makes them difficult to modify. There is now quite a bit of code for things like:

- automatic import of the program committee
- automatic import of collaborators for conflict identification.
- support for producing front matter for Springer LNCS volumes from accepted papers
- integration with our drag-n-drop editor for producing a program from the list of accepted papers
- support for two kinds of copyright forms
- custom submission fields for our workflow.
- external support for upload of videos, slides, and final versions of papers to integrate with our metadata store, websites, and youtube channel.
- an external API to update paper settings

**Comment by kohler** [comment]
I assume this is for IACR?

Many of these features would be very nice to integrate into the open-source version. Are you interested/able to share them?

Some of the features, such as drag-and-drop editing, are mostly used on HotCRP.com, but use broadly-available features, like tag orders. It would be nice to do something standard there.

**Comment by kmccurley** [comment]
yes, it's for IACR. I assume you have also built extensions for Usenix and ACM.

I had originally planned to incorporate these features into HotCRP itself, but they all fell by the wayside for a variety of reasons. The collaborator import was originally going to use DBLP, but I gave up on their API because it didn't seem to export the things I needed. I switched to using the iacr.org data store, which only covers cryptography publications but that makes it useless for other conferences.

The program editor also has dependencies on IACR authentication and the format used for the IACR websites. The LNCS stuff is perhaps better suited to the open source version, but it's very brittle and untested (the documentation from Springer is also pretty miserable). The LNCS preparation uses a drag-n-drop interface built using  [dragula](https://bevacqua.github.io/dragula/). Unfortunately COVID has more than doubled my workload for running virtual conferences, so I don't really have time to work on this.
